### PR TITLE
Fix reading from sparse columns after restart

### PR DESCRIPTION
**Pull request by CurtizJ**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fixed reading from sparse columns after changing setting `ratio_of_defaults_for_sparse_serialization` back to `1.0` and restart of server.

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
This is an automated comment for commit 0612f5d7820e70921748353efcea600cab0581ab with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/49660/0612f5d7820e70921748353efcea600cab0581ab/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: src/Storages/MergeTree/IMergeTreeDataPart.cpp [hunk 1]
```diff
@@ -1341,11 +1341,11 @@ void IMergeTreeDataPart::loadColumns(bool require)
         .choose_kind = false,
     };
 
-    SerializationInfoByName infos(loaded_columns, settings);
+    SerializationInfoByName infos;
     if (metadata_manager->exists(SERIALIZATION_FILE_NAME))
```
Comment (kssenii) [inline src/Storages/MergeTree/IMergeTreeDataPart.cpp:1345]: is it intended that now if in line 1345 condition is not satisfied then infos is not initialised?

Comment (CurtizJ) [reply src/Storages/MergeTree/IMergeTreeDataPart.cpp:1345]: Yes, it's ok. If `SerializationInfoByName` is not initialized the default serialization will be returned for all columns. It's what expected.

**Review by kssenii — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
The test `column ttl` has failed.

**Pull request merged**